server/blob/blob-export-file/src/main/java/org/apache/james/blob/export/file/LocalFileBlobExportMechanism.java

```diff
@@ -46,14 +50,46 @@
 
     public static class Configuration {
 
+        public static Configuration from(org.apache.commons.configuration.Configuration propertiesConfiguration) {
+            String exportDirectory = propertiesConfiguration.getString(DIRECTORY_LOCATION_PROPERTY);
+            return of(exportDirectory);
+        }
+
+        @VisibleForTesting
+        static Configuration of(String exportDirectory) {
```

Comment:
a method that creates a Configuration from a String looks like a bad idea.
* you can use a type (File)
* you can name the method (for example `usingExportDirectory`)

Reply:
with a same input string, directory created by `new File(inputString)` and `fileSystem.getFile(inputString)` maybe different. I think I need to create a Configuration.Factory to carries a FileSystem instance to produce the right Configuration with the right directory. Is that okay?

Reply:
We found that, by depending on FileSystem, the static default configuration can not be created because it requires FileSystem at compile time. FYI, @chibenwa is proposing a strong typing FileUrl class to replace plain file path string everywhere. I will use it to resolve you comment!

Reply:
see #2282 
I will leave this concern to be handle in another PR and continue making progress on this PR.

Reply:
could you at least rename the method as suggested?